### ISSUE TITLE
XmlSerializer Creation throws FileNotFoundException

### ISSUE THREAD

**Issue by david-maw** [dotnet/maui#8322]
### Description

The unexpected pause does not do any harm in that the missing file seems to be optional and the exception gets caught before it returns to user code, Nonetheless it's mighty confusing and tedious to get around, especially on Android, because even if you disable break on FileNotFoundException the debugger still pauses with a message like this:

![nobreakdialog](https://user-images.githubusercontent.com/12423046/175719472-6873b046-c12f-43b0-a329-1987415d87cb.jpg)

Even more annoyingly it seems to break (or pause) twice for each new XmlSerializer created.

There are several things that don't seem right:

1. Using an exception to report a normal situation (a missing optional file) is likely to cause confusion and is especially annoying migrating code from Xamarin which does not do this.
2. Throwing the exception once per XmlSerializer is bad enough, doing it twice is worse
3. In Android disabling that Break applies to all modules (unlike Windows where you can at least ignore exceptions in a named module). This may be a feature of Android.
4. When you DO inhibit the break in Android it pauses anyway (with the message above). 

If this needs to be split into multiple issues, I can do that, just let me know.  

### Steps to Reproduce

1. Unzip the project from [XmlSerializerException.zip](https://github.com/dotnet/maui/files/8982430/XmlSerializerException.zip)
2. In visual Studio Exception Settings enable Break on `System.IO.FileNotFoundException`
3.  Build and run the project on Android, it should break on the statement 
    `new XmlSerializer(typeof(Person)); ` 
    in MainPage.Xaml.cs
4. Press "continue" and it will break again.
5. Restart the application
6. This time disable Break on `System.IO.FileNotFoundException`
7. Run the project under the debugger again and note that it still pauses on  `System.IO.FileNotFoundException` even though Break is disabled for that Exception.  

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows

### Affected platform versions

Android 11, Windows 10 

### Did you find any workaround?

Just continue through it and the system will usually keep working.

**Comment by PureWeen** [comment]
@david-maw can you please upload your reproduction to a github repository?

**Comment by david-maw** [comment]
@PureWeen yes, but it'll be a little effort, is the zipped version not helpful? Should I submit future issues with GitHub repos rather than simply zipping up the sources? If so, that's fine, I can do it, it's just a difference in how I create examples.

**Comment by kristinx0211** [comment]
@david-maw I've tried run with your repro project, I can see the exception break twice, but when disable the `System.IO.FileNotFoundException` In visual Studio Exception Settings and restart the project, the `System.IO.FileNotFoundException` is not throw, see my video, tell me if I miss something.
![8322](https://user-images.githubusercontent.com/99457000/175918408-1dbc021b-7184-4e95-b54b-f30db9a4c94f.gif)

**Comment by david-maw** [comment]
@kristinx0211 Interesting, I see the same problem on two different Android versions (one in the emulator one a physical device), I wonder if it is something in my local configuration? I've reloaded VS and the Android SDK so if it is something local it's a bit obscure and of course, your question only addresses item 4, not the first three (which really boil down to "throwing an exception for a non-problem is bad").

Anyway, here's what I see. 
![XmlSerializerException](https://user-images.githubusercontent.com/12423046/175979000-17697bad-403f-4cba-a7cf-69d202b5a477.gif)

**Comment by rmarinho** [comment]
@david-maw  do you have break in all exceptions on? `https://www.christianfindlay.com/blog/visual-studio-break-on-all-exceptions`

**Comment by david-maw** [comment]
Hi @rmarinho all the CLR Exceptions **except** `System.IO.FileNotFoundException` are enabled, it is disabled, but it still breaks, which is just plain weird and perhaps some sort of local problem, interestingly when I turn off **all** CLR exceptions it no longer breaks.

As the URL you pointed to above says "if all is well, the app shouldn't throw exceptions" and indeed on Windows it doesn't, but it does in Android and that's my underlying complaint. Breaking on a disabled exception seems like it might be a separate issue.

**Comment by tipa** [comment]
I am having the same problem, however in .NET for Android (not using MAUI)

**Comment by mfeingol** [comment]
I'm observing this as well on Android and it is extremely annoying. We need to either be able to disable exceptions (disabling Exception and FileNotFoundException in Exception Settings does nothing), or XmlSerializer needs a way to be initialized without throwing all those exceptions. The type it's being given is available inside my assembly. There's no reason XmlSerializer needs to be attempting to load anything else.

**Comment by david-maw** [comment]
Certainly, the standard practice I'm familiar with is "don't throw exceptions if there's no error" and Microsoft's [Framework Design Guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/exception-throwing) say "DO NOT use exceptions for the normal flow of control, if possible" which seems pretty unambiguous. My money's on a simple oversight in the code but since I don't know the code, what looks trivial (check for a file being present before trying to open it) could be complex to change.
